test/kleros/kleros-liquid-extra-views.js

```diff
@@ -0,0 +1,262 @@
+    subcourtMap[subcourtTree.ID] = {
+      ...subcourtTree,
+      children:
+        subcourtTree.children && subcourtTree.children.map(child => child.ID)
+    }
+  }
+  return { subcourtMap, subcourtTree }
+}
+const asyncForEach = async (method, iterable) => {
+  const array = Array.isArray(iterable) ? iterable : Object.values(iterable)
+  for (const item of array) await method(item)
+}
+
+contract('xKlerosLiquid', accounts => {
+  let pinakion
+  let RNG
+  let governor
+  let minStakingTime
+  let maxDrawingTime
+  let subcourtTree
+  let subcourtMap
+  let klerosLiquid
+  let extraViews
+  beforeEach(async () => {
+    governor = accounts[0]
+    // Deploy contracts and generate subcourts
+    pinakion = await Pinakion.new(
+      0x0, // _tokenFactory
+      0x0, // _parentToken
+      0, // _parentSnapShotBlock
+      'Pinakion', // _tokenName
+      18, // _decimalUnits
+      'PNK', // _tokenSymbol
+      true // _transfersEnabled
+    )
+    await pinakion.generateTokens(governor, -1)
+
+    const randomNumber = 10
+    RNG = await ConstantNG.new(randomNumber)
+    minStakingTime = 1
+    maxDrawingTime = 1
+    const {
+      subcourtMap: _subcourtMap,
+      subcourtTree: _subcourtTree
+    } = generateSubcourts(randomInt(4, 2), 3)
+    subcourtTree = _subcourtTree
+    subcourtMap = _subcourtMap
+
+    klerosLiquid = await KlerosLiquid.new(
+      governor,
+      pinakion.address,
+      RNG.address,
+      minStakingTime,
+      maxDrawingTime,
+      subcourtTree.hiddenVotes,
+      subcourtTree.minStake,
+      subcourtTree.alpha,
+      subcourtTree.jurorFee,
+      subcourtTree.jurorsForJump,
+      subcourtTree.timesPerPeriod,
+      subcourtTree.sortitionSumTreeK
+    )
+
+    await pinakion.changeController(klerosLiquid.address)
+    extraViews = await KlerosLiquidExtraViews.new(klerosLiquid.address)
+  })
+
+  it('Should query delayed stakes correctly.', async () => {
+    await klerosLiquid.createSubcourt(
+      subcourtTree.parent,
+      subcourtTree.hiddenVotes,
+      subcourtTree.minStake + 1,
+      subcourtTree.alpha,
+      subcourtTree.jurorFee,
+      subcourtTree.jurorsForJump,
+      subcourtTree.timesPerPeriod,
+      subcourtTree.sortitionSumTreeK
+    )
+
+    const extraData = `0x${(0).toString(16).padStart(64, '0')}${(1)
+      .toString(16)
+      .padStart(64, '0')}`
+    await klerosLiquid.createDispute(2, extraData, {
+      value: await klerosLiquid.arbitrationCost(extraData)
+    })
+    await increaseTime(minStakingTime)
+    await klerosLiquid.passPhase()
+
+    await klerosLiquid.setStake(0, subcourtTree.minStake, {
+      from: governor
+    })
+    await klerosLiquid.setStake(1, subcourtTree.minStake + 1, {
+      from: governor
+    })
+
+    const stake0 = await extraViews.stakeOf(governor, 0)
+    const stake1 = await extraViews.stakeOf(governor, 1)
+    expect(stake0).to.eql(web3.toBigNumber(subcourtTree.minStake))
+    expect(stake1).to.eql(web3.toBigNumber(subcourtTree.minStake + 1))
+
```

Comment:
This test case could be split in two to improve readability.
I'd suggest to create a nested suite with `describe` like this:

```javascript
describe('When there are delayed stakes', () => {
  beforeEach('Setup delayed stakes', () => {
    // ...
  });

  it('Should get the total amount of tokens staked in a subcourt', () => {
    // Tests for .stakeOf()
  });

  it('Should get the proper juror stats', () => {
    // Tests for .getJuror()
  });
});
```

Reply:
Yes, I changed it.